### ISSUE TITLE
Enforce typing support

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#545]
Here are some typing rules that [are missing right now](https://github.com/wemake-services/wemake-python-styleguide/blob/master/setup.cfg#L68):

```python
disallow_untyped_decorators = True
no_implicit_optional = True
```

We need to refactor that.

**Comment by ripsandy** [comment]
Can I work on this?

**Comment by sobolevn** [comment]
@ripsandy you are welcome!

**Comment by sobolevn** [comment]
Done, will be shipped with `0.9.0`
